CMakeLists.txt

```diff
@@ -644,6 +729,20 @@ if(USE_CUDA)
   target_link_libraries(_lightgbm PRIVATE ${histograms})
 endif()
 
+if(USE_ROCM)
+  # histograms are list of object libraries. Linking object library to other
+  # object libraries only gets usage requirements, the linked objects won't be
+  # used. Thus we have to call target_link_libraries on final targets here.
+  if(BUILD_CLI)
+    target_link_libraries(lightgbm PRIVATE ${histograms})
+  endif()
+  target_link_libraries(_lightgbm PRIVATE ${histograms})
+endif()
+
+if(USE_HDFS)
```

Comment:
Remove this.

Reply:
Done in [8f6600e](https://github.com/microsoft/LightGBM/pull/6086/commits/8f6600edd7d5336d448c9e49b084491a10d7f501).